python/ribasim/ribasim/model.py

```diff
@@ -61,13 +64,30 @@
     xugrid = MissingOptionalModule("xugrid")
 
 
+class NodeID(BaseModel):
+    node_ids: set[int] = set()
+    max_node_id: NonNegativeInt = 0
```

Comment:
Could you add a small docstring explaining that this tracks the used and highest used node ID? Also consider renaming the class and Model field to something like `UsedNodeIDs`, because I thought it represented an actual node ID.